SimExLite/PropagationCalculators/WPGProgagationCalculator.py

```diff
@@ -0,0 +1,112 @@
+""":module WPGPropagationCalculator: Module that holds the WPGPropagationCalculator class."""
+
+from pathlib import Path
+import shutil
+import sys
+
+# Requires WPG package: https://github.com/samoylv/WPG
+from s2e.prop import propagate_s2e
+
+from libpyvinyl import BaseCalculator, CalculatorParameters
+from libpyvinyl.BaseData import DataCollection
+from SimExLite.WavefrontData import WavefrontData, WPGFormat
+
+
+class WPGPropagationCalculator(BaseCalculator):
+    """:class WPGPropagationCalculator: Class representing photon propagation through X-ray optics."""
+
+    def __init__(
+        self,
+        name: str,
+        input: DataCollection,
+        output_keys: str = "WPG_wavefront",
+        output_data_types=WavefrontData,
+        output_filenames: str = "wavefront.h5",
+        instrument_base_dir="./",
+        calculator_base_dir="WPGPropagationCalculator",
+        parameters=None,
+    ):
+        super().__init__(
+            name,
+            input,
+            output_keys,
+            output_data_types=output_data_types,
+            output_filenames=output_filenames,
+            instrument_base_dir=instrument_base_dir,
+            calculator_base_dir=calculator_base_dir,
+            parameters=parameters,
+        )
+        self.__WPG_path = None
```

Comment:
still needed? above, `from s2e import ...` already requires WPG to be properly installed.

Reply:
Deleted. Then implement the similar way as GSC.